internal/command/deploy/machines_deploymachinesapp.go

```diff
@@ -79,6 +79,26 @@ func (md *machineDeployment) restartMachinesApp(ctx context.Context) error {
 	return md.updateExistingMachines(ctx, machineUpdateEntries)
 }
 
+func (md *machineDeployment) inferCanaryGuest(name string) *api.MachineGuest {
```

Comment:
Just a personal one but this would read easier to me if it was something like

```go
mInGroup := lo.Filter(machines, machine.ProcessGroup() == name)
mInGroup[0].Config.Guest
```

I think this achieves the same thing (at the expense of having to iterate over everything)